### PR TITLE
Making clippy happy

### PR DESCRIPTION
**Pull request by yaleman**
Fixes a load of `cargo clippy` complaints. 

Replaces the use of the string username `"admin"` in tests with a single common variable - same as the password is for same tests.

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

All the above tests were run on rust 1.49.0 across:

 - openSUSE Tumbleweed
 - openSUSE Leap 15.2
 - Ubuntu Bionic 
 - macOS Big Sur on M1
 - Debian Buster

### CODE REVIEW

**Comment by yaleman** [comment]
🔥 #420 blazeit

**Comment by yaleman** [comment]
... test failed? That's rude :(

**Comment by Firstyear** [comment]
Test failure is a known random issue in tests.

File: kanidmd/src/lib/lib.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #![deny(clippy::unwrap_used)]
 #![deny(clippy::expect_used)]
 // #![deny(clippy::panic)]
-#![deny(clippy::unreachable)]
+#![allow(clippy::unreachable)]
```
Comment (Firstyear) [inline kanidmd/src/lib/lib.rs:6]: We should deny unreachhable an annotate individual locations where we need to allow it I think.

File: kanidmd_web_ui/src/login.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@ extern "C" {
 pub struct LoginApp {
     link: ComponentLink<Self>,
     inputvalue: String,
+    #[allow(dead_code)]
```
Comment (Firstyear) [inline kanidmd_web_ui/src/login.rs:23]: Shouldn't be needed, lstorage should be used in the latest commits.

File: kanidmd_web_ui/src/login.rs [hunk 3]
```diff
@@ -262,7 +263,7 @@ impl Component for LoginApp {
     type Properties = ();
 
     fn create(_: Self::Properties, link: ComponentLink<Self>) -> Self {
-        ConsoleService::log(format!("create").as_str());
+        ConsoleService::log(&"create".to_string());
```
Comment (Firstyear) [inline kanidmd_web_ui/src/login.rs:266]: Should just be "create" no need to & and to_string here.

**Comment by Firstyear** [comment]
@yaleman Do you mind if I merge #419 first? I think it may need you to rebase though. Alternately if you do these little fix ups I'll merge this first, then I'll rebase 419 to this change. Which ever is easier :)

File: kanidmd_web_ui/src/login.rs (hunk 2 above)
Comment (yaleman) [reply kanidmd_web_ui/src/login.rs:23]: done

**Comment by yaleman** [comment]
@Firstyear go ahead with #419  - happy to rebase :)

File: kanidmd_web_ui/src/login.rs (hunk 3 above)
Comment (yaleman) [reply kanidmd_web_ui/src/login.rs:266]: done

**Pull request merged**